packages/core-ethereum/crates/core-ethereum-indexer/src/handlers.rs

```diff
@@ -1409,23 +1425,16 @@ pub mod wasm {
         pub fn init(
             address_to_monitor: &str,
             chain_key: &str,
-            contract_addresses: ContractAddresses,
+            contract_addresses_js: JsValue,
             callbacks: IndexerCallbacks,
         ) -> Handlers {
+            let get_contract_info = serde_wasm_bindgen::from_value::<crate::handlers::wasm::ContractAddresses>(contract_addresses_js).unwrap();
+            let contract_addresses = super::ContractAddresses::from(&get_contract_info);
             Self {
                 w: super::ContractEventHandlers {
                     address_to_monitor: Address::from_str(address_to_monitor).unwrap(),
                     chain_key: Address::from_str(chain_key).unwrap(),
-                    addresses: super::DeploymentExtract {
-                        channels: Address::from_str(contract_addresses.channels().as_str()).unwrap(),
-                        token: Address::from_str(contract_addresses.token().as_str()).unwrap(),
-                        network_registry: Address::from_str(contract_addresses.network_registry().as_str()).unwrap(),
-                        node_safe_registry: Address::from_str(contract_addresses.node_safe_registry().as_str())
-                            .unwrap(),
-                        announcements: Address::from_str(contract_addresses.announcements().as_str()).unwrap(),
-                        node_management_module: Address::from_str(contract_addresses.node_management_module().as_str())
-                            .unwrap(),
-                    },
+                    addresses: contract_addresses,
```

Comment:
You could also do `get_contract_info.into()`

Reply:
Thank you! Updated it.